### PR TITLE
Pagination basic version

### LINKED ISSUE #267: Set up pagination for list view
The List View should have continuous scrolling that displays a certain amount initially and as the user scrolls down the list more "pages" are fetched so that more data is available.

### PR DESCRIPTION
**Pull request by harrybabu**
Fixes #267 

Basic version, not using json-api though.

### CODE REVIEW

**Comment by nimishamukherjee** [comment]
@harrybabu we need to run re-install? npm start gives error - Cannot find module 'angular2-infinite-scroll'.

**Comment by harrybabu** [comment]
@nimishamukherjee added new node module named 'angular2-infinite-scroll' npm install should work.

File: src/app/work-item/work-item.service.ts [hunk 1]
```diff
@@ -16,25 +16,46 @@ export class WorkItemService {
   private workItemTypeUrl = process.env.API_URL + 'workitemtypes';
   private availableStates: DropdownOption[] = [];
   public workItemTypes: WorkItemType[] = [];
+  private isSearching: Boolean = false;
+  private nextLinks: Object = {};
```
Comment (aslakknutsen) [inline src/app/work-item/work-item.service.ts]: Should probably be called just 'links'. nextLinks is just one of the many possible links. And not all links would be related to pagination.

Comment (harrybabu) [inline src/app/work-item/work-item.service.ts]: Got it 👍

File: src/app/work-item/work-item.service.spec.ts [hunk 2]
```diff
@@ -64,25 +64,29 @@ describe('Work Item Service - ', () => {
...
+          'system.title': 'test1'
+        },
+        'id': '1',
+        'type': 'system.userstory',
+        'version': 0
+      } 
   ] as WorkItem[];
 
+  let responseData = {
+    'data': response
```
Comment (harrybabu) [inline src/app/work-item/work-item.service.spec.ts]: Not using links in the current unit test case, will be updating it in future.

**Comment by aslakknutsen** [comment]
It's a little bit excessive to fetch the next set of data for each time the scroll bar move, ever so slightly.

**Comment by harrybabu** [comment]
@aslakknutsen absolutely, in the radar working on it...

**Comment by joshuawilson** [comment]
@harrybabu please rebase

**Comment by harrybabu** [comment]
@joshuawilson rebased

**Comment by joshuawilson** [comment]
This was completed in PR #424. Closing.

**Pull request closed without merging**